lib/matplotlib/axes/_base.py

```diff
@@ -146,12 +147,17 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__ = state.copy()
-        self.set_color_cycle()
-
-    def set_color_cycle(self, clist=None):
-        if clist is None:
-            clist = rcParams['axes.color_cycle']
-        self.color_cycle = itertools.cycle(clist)
+        self.set_prop_cycle()
+
+    def set_prop_cycle(self, prop_cycler=None):
+        if prop_cycler is None:
+            prop_cycler = rcParams['axes.prop_cycle']
+            if prop_cycler is None and 'axes.color_cycle' in rcParams:
+                clist = rcParams['axes.color_cycle']
+                prop_cycler = cycler('color', clist)
+        self.prop_cycler = itertools.cycle(prop_cycler)
+        # Make a copy
+        self._prop_keys = list(prop_cycler.keys)
```

Comment:
This also reminded me I needed to open https://github.com/matplotlib/cycler/pull/7